diff --git a/vendor/hyva-themes/magento2-hyva-checkout-braintree/view/frontend/templates/checkout/payment/method/card.phtml b/vendor/hyva-themes/magento2-hyva-checkout-braintree/view/frontend/templates/checkout/payment/method/card.phtml
index 9a93750..65919bc 100644
--- a/vendor/hyva-themes/magento2-hyva-checkout-braintree/view/frontend/templates/checkout/payment/method/card.phtml
+++ b/vendor/hyva-themes/magento2-hyva-checkout-braintree/view/frontend/templates/checkout/payment/method/card.phtml
@@ -374,6 +374,7 @@ $availableCardTypes = $cardViewModel->getAvailableCardTypes();
                         throw new Error('<?= $escaper->escapeJs(__("3DS validation failed.")) ?>');
                     }
 
+                    await Magewire.find('checkout.payment.method.' + 'signifyd').call('setCardData', response.details);
                     await Magewire.find('checkout.payment.method.' + braintreeId).call('setIsVaulted', vault);
                     await Magewire.find('checkout.payment.method.' + braintreeId).call('setToken', response.nonce);
                     clearError();
