From efa1fe040b7a29085a14beaa5b7cfa416c77537e Mon Sep 17 00:00:00 2001
From: ebanolopes <ebano@onbi.com.br>
Date: Tue, 5 May 2026 15:18:28 +0000
Subject: [PATCH] MAG-995: Webhook processing failing to unhold and returning
 200

---
 Controller/Webhooks/Index.php | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/vendor/signifyd/module-connect/Controller/Webhooks/Index.php b/vendor/signifyd/module-connect/Controller/Webhooks/Index.php
index 5d6aceb..895b9c2 100644
--- a/vendor/signifyd/module-connect/Controller/Webhooks/Index.php
+++ b/vendor/signifyd/module-connect/Controller/Webhooks/Index.php
@@ -409,9 +409,18 @@ class Index implements HttpPostActionInterface
 
             $updateOrder = $this->updateOrderFactory->create();
             $case = $updateOrder($case);
-
             $this->casedataResourceModel->save($case);
 
+            if ($case->getEntries('fail') == 1 && $case->getMagentoStatus() !== Casedata::COMPLETED_STATUS) {
+                $this->logger->error(
+                    "WEBHOOK: Failed to update case {$caseId}",
+                    ['entity' => $case]
+                );
+                $result->setData(['message' => 'Webhook processing failed']);
+                $result->setHttpResponseCode(Http::STATUS_CODE_500);
+                return $result;
+            }
+
             $result->setHttpResponseCode(Http::STATUS_CODE_200);
             return $result;
         } catch (\Exception $e) {
-- 
2.25.1

