From 5949a1e0ab605f97d1e0c4b8858e71666b5fb369 Mon Sep 17 00:00:00 2001
From: ebanolopes <ebano@onbi.com.br>
Date: Tue, 10 Dec 2024 19:56:56 +0000
Subject: [PATCH] MAG-942: Review Adyen integration for 9.4.0 and up

---
 Model/ConfigProvider.php | 52 +++++++++++++++++++++++++++++++++++++++-
 1 file changed, 51 insertions(+), 1 deletion(-)

diff --git a/vendor/signifyd/module-connect/Model/ConfigProvider.php b/vendor/signifyd/module-connect/Model/ConfigProvider.php
index 1e1d928..51666dc 100644
--- a/vendor/signifyd/module-connect/Model/ConfigProvider.php
+++ b/vendor/signifyd/module-connect/Model/ConfigProvider.php
@@ -11,6 +11,9 @@ namespace Signifyd\Connect\Model;
 use Magento\Store\Model\StoreManagerInterface;
 use Signifyd\Connect\Helper\ConfigHelper;
 use Magento\Framework\Module\ModuleListInterface;
+use Magento\Framework\Component\ComponentRegistrarInterface;
+use Magento\Framework\Component\ComponentRegistrar;
+use Magento\Framework\Filesystem\Directory\ReadFactory;
 
 class ConfigProvider implements \Magento\Checkout\Model\ConfigProviderInterface
 {
@@ -30,11 +33,27 @@ class ConfigProvider implements \Magento\Checkout\Model\ConfigProviderInterface
     public $moduleListInterface;
 
     /**
+     * @var ComponentRegistrarInterface
+     */
+    public $componentRegistrar;
+
+    /**
+     * @var ReadFactory
+     */
+    public $readFactory;
+
+    /**
+     * ConfigProvider constructor.
+     *
+     * @param ReadFactory $readFactory
+     * @param ComponentRegistrarInterface $componentRegistrar
      * @param ConfigHelper $configHelper
      * @param StoreManagerInterface $storeManager
      * @param ModuleListInterface $moduleListInterface
      */
     public function __construct(
+        ReadFactory $readFactory,
+        ComponentRegistrarInterface $componentRegistrar,
         ConfigHelper $configHelper,
         StoreManagerInterface $storeManager,
         ModuleListInterface $moduleListInterface
@@ -42,7 +61,16 @@ class ConfigProvider implements \Magento\Checkout\Model\ConfigProviderInterface
         $this->storeManager = $storeManager;
         $this->moduleListInterface = $moduleListInterface;
         $this->configHelper = $configHelper;
+        $this->componentRegistrar = $componentRegistrar;
+        $this->readFactory = $readFactory;
     }
+
+    /**
+     * Get config method
+     *
+     * @return array[]
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
     public function getConfig()
     {
         $policyName = $this->configHelper->getPolicyName(
@@ -55,7 +83,7 @@ class ConfigProvider implements \Magento\Checkout\Model\ConfigProviderInterface
         $adyenModule = $this->moduleListInterface->getOne('Adyen_Payment');
 
         if (isset($adyenModule)) {
-            $adyenVersion = $this->moduleListInterface->getOne('Adyen_Payment')['setup_version'];
+            $adyenVersion = $this->getModuleVersionFromComposer('Adyen_Payment');
             $isAdyenGreaterThanEightEighteen = version_compare($adyenVersion, '8.18.0') >= 0;
             $isAdyenGreaterThanEight = version_compare($adyenVersion, '8.0.0') >= 0 &&
                 version_compare($adyenVersion, '8.17.9') <= 0;
@@ -74,4 +102,26 @@ class ConfigProvider implements \Magento\Checkout\Model\ConfigProviderInterface
             'isAdyenGreaterThanEight' => $isAdyenGreaterThanEight]
         ];
     }
+
+    /**
+     * Get version from composer.json if it exists otherwise return empty string
+     *
+     * @param string $moduleName
+     * @return string
+     */
+    public function getModuleVersionFromComposer(string $moduleName) : string
+    {
+        $path = $this->componentRegistrar->getPath(ComponentRegistrar::MODULE, $moduleName);
+        $directoryRead = $this->readFactory->create($path);
+        $composerJsonData = $directoryRead->readFile('composer.json');
+
+        $data = json_decode($composerJsonData, true);
+
+        if (is_null($data))
+        {
+            return '';
+        }
+
+        return $data['version']?? '';
+    }
 }
-- 
2.25.1

