diff --git a/vendor/signifyd/module-connect/Controller/Webhooks/Index.php b/vendor/signifyd/module-connect/Controller/Webhooks/Index.php
index 4106b41..ac4f3b1 100644
--- a/vendor/signifyd/module-connect/Controller/Webhooks/Index.php
+++ b/vendor/signifyd/module-connect/Controller/Webhooks/Index.php
@@ -295,7 +295,11 @@ class Index extends Action
             ) {
                 $httpCode = Http::STATUS_CODE_400;
                 throw new LocalizedException(__("Case {$caseId} it is not ready to be updated"));
-            } elseif ($case->getPolicyName() === Casedata::PRE_AUTH) {
+            } elseif (
+                $case->getPolicyName() === Casedata::PRE_AUTH &&
+                $case->getGuarantee() !== 'HOLD' &&
+                $case->getGuarantee() !== 'PENDING'
+            ) {
                 $httpCode = Http::STATUS_CODE_200;
                 throw new LocalizedException(
                     __("Case {$caseId} already completed by synchronous response, no action will be taken")
